skipper/runner.py

```diff
@@ -89,14 +89,27 @@ def _run_nested(fqdn_image, environment, command, interactive, name, net, volume
 
 def handle_volumes_bind_mount(docker_cmd, homedir, volumes, workspace):
     volumes = volumes or []
-    volumes.extend([
-        '%(workspace)s:%(workspace)s:rw,Z' % dict(workspace=workspace),
-        '%(homedir)s/.netrc:%(homedir)s/.netrc:ro' % dict(homedir=homedir),
-        '%(homedir)s/.gitconfig:%(homedir)s/.gitconfig:ro' % dict(homedir=homedir),
-        '/var/lib/osmosis:/var/lib/osmosis:rw,Z',
-        '/var/run/docker.sock:/var/run/docker.sock:Z',
-        '%s:/opt/skipper/skipper-entrypoint.sh:Z' % utils.get_extra_file("skipper-entrypoint.sh"),
-    ])
+    if utils.get_runtime_command() == utils.PODMAN:
+        volumes.extend([
+            '%(workspace)s:%(workspace)s:rw,shared' % dict(workspace=workspace),
+            '%(homedir)s/.netrc:%(homedir)s/.netrc:ro' % dict(homedir=homedir),
+            '%(homedir)s/.gitconfig:%(homedir)s/.gitconfig:ro' % dict(homedir=homedir),
+            '%s:/opt/skipper/skipper-entrypoint.sh:rw' % utils.get_extra_file("skipper-entrypoint.sh"),
+        ])
+        if os.path.exists('/var/run/docker.sock'):
+            volumes.append('/var/run/docker.sock:/var/run/docker.sock:rw')
+        if os.path.exists('/var/lib/osmosis'):
+            volumes.append('/var/lib/osmosis:/var/lib/osmosis:rw')
+    else:
+        volumes.extend([
+            '%(workspace)s:%(workspace)s:rw,Z' % dict(workspace=workspace),
+            '%(homedir)s/.netrc:%(homedir)s/.netrc:ro' % dict(homedir=homedir),
```

Comment:
Some of these seems the same for all cases.
start a volumes array with them and than extend it in the if/else

Reply:
yap right